{"parents": [{"link": "../", "title": "Release notes"}], "prev": {"link": "../2.2.17/", "title": "Django 2.2.17 release notes"}, "next": {"link": "../2.2.15/", "title": "Django 2.2.15 release notes"}, "title": "Django 2.2.16 release notes", "meta": {}, "body": "<div class=\"section\" id=\"s-django-2-2-16-release-notes\">\n<span id=\"django-2-2-16-release-notes\"></span><h1>Django 2.2.16 release notes<a class=\"headerlink\" href=\"#django-2-2-16-release-notes\" title=\"Permalink to this headline\">\u00b6</a></h1>\n<p><em>September 1, 2020</em></p>\n<p>Django 2.2.16 fixes two security issues and two data loss bugs in 2.2.15.</p>\n<div class=\"section\" id=\"s-cve-2020-24583-incorrect-permissions-on-intermediate-level-directories-on-python-3-7\">\n<span id=\"cve-2020-24583-incorrect-permissions-on-intermediate-level-directories-on-python-3-7\"></span><h2>CVE-2020-24583: Incorrect permissions on intermediate-level directories on Python 3.7+<a class=\"headerlink\" href=\"#cve-2020-24583-incorrect-permissions-on-intermediate-level-directories-on-python-3-7\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<p>On Python 3.7+, <a class=\"reference internal\" href=\"../../ref/settings/#std-setting-FILE_UPLOAD_DIRECTORY_PERMISSIONS\"><code class=\"xref std std-setting docutils literal notranslate\"><span class=\"pre\">FILE_UPLOAD_DIRECTORY_PERMISSIONS</span></code></a> mode was not\napplied to intermediate-level directories created in the process of uploading\nfiles and to intermediate-level collected static directories when using the\n<a class=\"reference internal\" href=\"../../ref/contrib/staticfiles/#django-admin-collectstatic\"><code class=\"xref std std-djadmin docutils literal notranslate\"><span class=\"pre\">collectstatic</span></code></a> management command.</p>\n<p>You should review and manually fix permissions on existing intermediate-level\ndirectories.</p>\n</div>\n<div class=\"section\" id=\"s-cve-2020-24584-permission-escalation-in-intermediate-level-directories-of-the-file-system-cache-on-python-3-7\">\n<span id=\"cve-2020-24584-permission-escalation-in-intermediate-level-directories-of-the-file-system-cache-on-python-3-7\"></span><h2>CVE-2020-24584: Permission escalation in intermediate-level directories of the file system cache on Python 3.7+<a class=\"headerlink\" href=\"#cve-2020-24584-permission-escalation-in-intermediate-level-directories-of-the-file-system-cache-on-python-3-7\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<p>On Python 3.7+, the intermediate-level directories of the file system cache had\nthe system\u2019s standard umask rather than <code class=\"docutils literal notranslate\"><span class=\"pre\">0o077</span></code> (no group or others\npermissions).</p>\n</div>\n<div class=\"section\" id=\"s-bugfixes\">\n<span id=\"bugfixes\"></span><h2>Bugfixes<a class=\"headerlink\" href=\"#bugfixes\" title=\"Permalink to this headline\">\u00b6</a></h2>\n<ul class=\"simple\">\n<li>Fixed a data loss possibility in the\n<a class=\"reference internal\" href=\"../../ref/models/querysets/#django.db.models.query.QuerySet.select_for_update\" title=\"django.db.models.query.QuerySet.select_for_update\"><code class=\"xref py py-meth docutils literal notranslate\"><span class=\"pre\">select_for_update()</span></code></a>. When using\nrelated fields pointing to a proxy model in the <code class=\"docutils literal notranslate\"><span class=\"pre\">of</span></code> argument, the\ncorresponding model was not locked (<a class=\"reference external\" href=\"https://code.djangoproject.com/ticket/31866\">#31866</a>).</li>\n<li>Fixed a data loss possibility, following a regression in Django 2.0, when\ncopying model instances with a cached fields value (<a class=\"reference external\" href=\"https://code.djangoproject.com/ticket/31863\">#31863</a>).</li>\n</ul>\n</div>\n</div>\n", "metatags": "", "rellinks": [["genindex", "General Index", "I", "index"], ["py-modindex", "Python Module Index", "", "modules"], ["releases/2.2.15", "Django 2.2.15 release notes", "N", "next"], ["releases/2.2.17", "Django 2.2.17 release notes", "P", "previous"]], "sourcename": "releases/2.2.16.txt", "toc": "<ul>\n<li><a class=\"reference internal\" href=\"#\">Django 2.2.16 release notes</a><ul>\n<li><a class=\"reference internal\" href=\"#cve-2020-24583-incorrect-permissions-on-intermediate-level-directories-on-python-3-7\">CVE-2020-24583: Incorrect permissions on intermediate-level directories on Python 3.7+</a></li>\n<li><a class=\"reference internal\" href=\"#cve-2020-24584-permission-escalation-in-intermediate-level-directories-of-the-file-system-cache-on-python-3-7\">CVE-2020-24584: Permission escalation in intermediate-level directories of the file system cache on Python 3.7+</a></li>\n<li><a class=\"reference internal\" href=\"#bugfixes\">Bugfixes</a></li>\n</ul>\n</li>\n</ul>\n", "display_toc": true, "page_source_suffix": ".txt", "current_page_name": "releases/2.2.16", "sidebars": ["localtoc.html", "relations.html", "sourcelink.html", "searchbox.html"], "customsidebar": null, "favicon_url": null, "logo_url": null, "alabaster_version": "0.7.13", "include_console_assets": false}